velox/common/memory/MemoryPool.cpp

```diff
@@ -991,10 +992,15 @@ void MemoryPoolImpl::abort(const std::exception_ptr& error) {
   if (reclaimer() == nullptr) {
     VELOX_FAIL("Can't abort the memory pool {} without reclaimer", name_);
   }
-  aborted_ = true;
+  setAbortError(error);
   reclaimer()->abort(this, error);
 }
 
+void MemoryPoolImpl::setAbortError(const std::exception_ptr& error) {
+  abortError_ = error;
```

Comment:
```
VELOX_CHECK(!aborted_);
```

Reply:
What if there are multiple threads trying to set at the same time?